packages/test-kit/src/sync-base-fs-contract.ts

```diff
@@ -409,6 +409,107 @@ export function syncBaseFsContract(testProvider: () => Promise<ITestInput<IBaseF
+
+                fs.writeFileSync(sourcePath, SAMPLE_CONTENT)
+                fs.mkdirSync(join(tempDirectoryPath, 'dir'))
+                fs.mkdirSync(join(tempDirectoryPath, 'dir', 'subdir'))
+
+                fs.renameSync(sourcePath, destinationPath)
+
+                expect(fs.statSync(destinationPath).isFile()).to.equal(true)
+                expect(fs.readFileSync(destinationPath)).to.eql(SAMPLE_CONTENT)
+                expect(() => fs.statSync(sourcePath)).to.throw('ENOENT')
+            })
+
+            it('moves a directory', () => {
+                const { fs, tempDirectoryPath } = testInput
+                const { join } = fs.path
+                const sourcePath = join(tempDirectoryPath, 'dir')
+                const destinationPath = join(tempDirectoryPath, 'anotherDir', 'subdir', 'movedDir')
+                fs.mkdirSync(join(tempDirectoryPath, 'dir'))
+                fs.mkdirSync(join(tempDirectoryPath, 'anotherDir'))
+                fs.mkdirSync(join(tempDirectoryPath, 'anotherDir', 'subdir'))
+                fs.writeFileSync(join(sourcePath, 'file'), SAMPLE_CONTENT)
+
+                fs.renameSync(sourcePath, destinationPath)
+
+                expect(fs.statSync(destinationPath).isDirectory()).to.equal(true)
+                expect(fs.readFileSync(join(destinationPath, 'file'))).to.eql(SAMPLE_CONTENT)
+                expect(() => fs.statSync(sourcePath)).to.throw('ENOENT')
+            })
+
+            it('throws if source path doesn\'t exist', () => {
+                const { fs, tempDirectoryPath } = testInput
+                const { join } = fs.path
+                const sourcePath = join(tempDirectoryPath, 'file')
+
+                expect(() => fs.renameSync(sourcePath, join(tempDirectoryPath, 'file2'))).to.throw('ENOENT')
+            })
+
+            it('throws if the containing directory of the source path doesn\'t exist', () => {
+                const { fs, tempDirectoryPath } = testInput
+                const { join } = fs.path
+                const sourcePath = join(tempDirectoryPath, 'unicorn', 'file')
+
+                expect(() => fs.renameSync(sourcePath, join(tempDirectoryPath, 'file2'))).to.throw('ENOENT')
+            })
+
+            it('throws if destination containing path doesn\'t exist', () => {
+                const { fs, tempDirectoryPath } = testInput
+                const { join } = fs.path
+                const sourcePath = join(tempDirectoryPath, 'file')
+
+                fs.writeFileSync(sourcePath, SAMPLE_CONTENT)
+
+                expect(() => fs.renameSync(sourcePath, join(tempDirectoryPath, 'file2'))).not.to.throw('ENOENT')
+                expect(() => fs.renameSync(sourcePath, join(tempDirectoryPath, 'dir', 'file2'))).to.throw('ENOENT')
+            })
+
+            it('doesn\'t throw if destination path already exists for copying a directory over a non-existing directory', () => { //tslint:disable-line
```

Comment:
This testing case defines the desired behavior, other tests may be changed, refactored or even removed, I feel it's good to be explicit about these things but I also understand how it might seem odd. The best way I'd say is to express such things with types whenever possible.